mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java

```diff
@@ -223,6 +224,7 @@ private void addMessageToMailboxes(MessageIdMapper messageIdMapper, MailboxMessa
         for (MailboxId mailboxId : mailboxIds) {
             SimpleMailboxMessage copy = SimpleMailboxMessage.copy(mailboxId, mailboxMessage);
             MessageMetaData metaData = save(mailboxSession, messageIdMapper, copy);
+            metaData.getFlags().add(Flag.RECENT);
```

Comment:
I don't understand why you don't save it as recent (why this line is not before save call) ?